src/main/java/projectbookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,81 @@
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import projectbookstore.dto.book.BookDtoWithoutCategoryIds;
+import projectbookstore.dto.category.CategoryDto;
+import projectbookstore.dto.category.CreateCategoryRequestDto;
+import projectbookstore.service.BookService;
+import projectbookstore.service.CategoryService;
+
+@Tag(name = "Category management", description = "Endpoints for managing categories")
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/api/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @GetMapping
+    @PreAuthorize("hasAnyRole('ROLE_USER','ROLE_ADMIN')")
+    @Operation(summary = "Get all categories", description = "Get list of all available categories")
+    public List<CategoryDto> getAll(Authentication authentication, Pageable pageable) {
+        return categoryService.findAll(pageable);
+    }
+
+    @GetMapping("/{id}/books")
+    @PreAuthorize("hasAnyRole('ROLE_USER','ROLE_ADMIN')")
+    @Operation(summary = "Get books of a certain category by category id",
+            description = "Get books of a certain category by category id")
+    public List<BookDtoWithoutCategoryIds> getBooksByCategoryId(
+            @PathVariable Long id, Pageable pageable) {
+        return bookService.findAllByCategoryId(id,pageable);
+    }
+
+    @GetMapping("/{id}")
+    @PreAuthorize("hasAnyRole('ROLE_USER','ROLE_ADMIN')")
+    @Operation(summary = "Get a category for id", description = "Get a category for id")
+    public CategoryDto getCategoryById(@PathVariable Long id) {
+        return categoryService.findById(id);
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @ResponseStatus(HttpStatus.CREATED)
+    @PostMapping
+    @Operation(summary = "Create a new category", description = "Create a new category")
+    public CategoryDto createCategory(@Valid @RequestBody CreateCategoryRequestDto requestDto) {
+        return categoryService.save(requestDto);
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("/{id}")
+    @Operation(summary = "Update the category", description = "Update the category")
+    public CategoryDto updateBook(@PathVariable Long id,
+                              @RequestBody CreateCategoryRequestDto requestDto) {
+        return categoryService.update(id,requestDto);
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
```

Comment:
if we use `hasRole()` instead of `hasAuthorithy()` it should be like that, and remove `ROLE_` in getter of enum if you have it there in order for it to work
```suggestion
    @PreAuthorize("hasRole('ADMIN')")
```